beam-cpp-example/pipeline.py

```diff
@@ -0,0 +1,155 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import argparse
+import logging
+import os
+import sys
+
+
+def run(argv):
+  # Import here to avoid __main__ session pickling issues.
+  import io
+  import itertools
+  import matplotlib.pyplot as plt
+  import collatz
+
+  import apache_beam as beam
+  from apache_beam.io import restriction_trackers
+  from apache_beam.options.pipeline_options import PipelineOptions
+
+  class RangeSdf(beam.DoFn, beam.RestrictionProvider):
+    """An SDF producing all the integers in the input range.
+
+    This is preferable to beam.Create(range(...)) as it produces the integers
```

Comment:
nit: This make sense, but it is making the primary example of using the cpp in beam more complicated as opposed to using a create function. (No need to change it really.)